### PR TITLE
Add Buoyant Cloud integration

### PR DESCRIPTION
**Pull request by siggy**
Signed-off-by: Andrew Seigner <siggy@buoyant.io>

### What does this PR do?

This PR adds the first version of the Buoyant Cloud integration. This integration is OAuth/API-based.

### Motivation

What inspired you to submit this pull request?

Our Buoyant Cloud customers have requested their data be forwarded to Datadog.

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
  - Note: Extensive tests already in place on the Buoyant Cloud side
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

File: buoyant_cloud/README.md
Comment (emarsha94) [inline buoyant_cloud/README.md:13]: Thanks for updating this! Could you include 1-2 sentences on the value of the integration and what it does?  Looks like that is the last part we need here

File: buoyant_cloud/README.md
Comment (emarsha94) [inline buoyant_cloud/README.md:20]:
```suggestion
You need to have an account at [Buoyant Cloud][1] to use this integration. You can also sign up for Buoyant Cloud in the Datadog Marketplace.
```

Comment (emarsha94) [reply buoyant_cloud/README.md:20]: fixed spacing

File: buoyant_cloud/README.md
Comment (emarsha94) [inline buoyant_cloud/README.md:30]:
```suggestion
5. Go to the `Destinations` section and select your Datadog account to send all events matching the notification rule sent to Datadog.
```

Comment (emarsha94) [reply buoyant_cloud/README.md:30]: Are the notification rules already set or is this something the customer needs to do on their end? If so, we should add it as a step

File: buoyant_cloud/README.md
Comment (emarsha94) [inline buoyant_cloud/README.md:35]:
```suggestion
As Buoyant Cloud creates events, you should see them appear in Datadog [event explorer][4].
```

Comment (emarsha94) [reply buoyant_cloud/README.md:35]: fixed spacing

File: buoyant_cloud/README.md
Comment (emarsha94) [inline buoyant_cloud/README.md:43]:
```suggestion
4. In the Datadog UI, remove any related Buoyant Cloud API keys from Organization Settings
```

**Review by emarsha94 — CHANGES_REQUESTED** [review]
Thanks for making the updates! Fixed some spacing and made some suggestions. You may know this already but you can commit the suggestions directly to fix the spacing :) Let me know if you have questions on this!

**Comment by siggy** [comment]
@emarsha94 Second round of feedback incorporated, thanks again.

> You may know this already but you can commit the suggestions directly to fix the spacing :) Let me know if you have questions on this!

Yup, I'm just a bit OCD about gathering all my changes on my side so I can read them in my editor (and also not cause multiple notifications sent reviewers). I do very much appreciate the very specific feedback though, thank you for that. 👍

File: buoyant_cloud/README.md
Comment (emarsha94) [inline buoyant_cloud/README.md:16]:
```suggestion
[Buoyant Cloud][1] provides fully managed Linkerd on your cluster to monitor the health of Linkerd and deployments. With this integration, you can monitor and be alerted on Linkerd's health, workload traffic, and rollout events.
```

File: buoyant_cloud/assets/dashboards/buoyant_cloud_overview.json
Comment (emarsha94) [inline buoyant_cloud/assets/dashboards/buoyant_cloud_overview.json:3]: Please include a description of the dashboard

**Review by emarsha94 — CHANGES_REQUESTED** [review]

File: buoyant_cloud/manifest.json
Comment (bgoldberg122) [inline buoyant_cloud/manifest.json:3]: I confirmed this matches the `published_app_uuid` of the developer platform app. Thanks for doing that step!

File: buoyant_cloud/README.md
Comment (bgoldberg122) [inline buoyant_cloud/README.md:50]: This should be a relative link so it works in all datacenters:
```suggestion
[4]: /event/explorer
```

**Review by bgoldberg122 — CHANGES_REQUESTED** [review]

File: buoyant_cloud/README.md
Comment (bgoldberg122) [inline buoyant_cloud/README.md:30]: We can remove this line, since it will be automatically added on our side :)

File: buoyant_cloud/README.md
Comment (buraizu) [inline buoyant_cloud/README.md:35]: Would it be possible to provide some additional details about these events?

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for the PR, I made a few suggestions for the docs

File: buoyant_cloud/README.md
Comment (buraizu) [inline buoyant_cloud/README.md:19]:
```suggestion
5. Go to the **Destinations** section and select your Datadog account to send all events matching the notification rule to Datadog.
```

File: buoyant_cloud/README.md
Comment (buraizu) [inline buoyant_cloud/README.md:55]:
```suggestion
[4]: https://app.datadoghq.com/event/explorer
```

Comment (buraizu) [reply buoyant_cloud/README.md:55]: I recognize that this change was requested by my colleague [here](https://github.com/DataDog/integrations-extras/pull/1693#discussion_r1049034864), but unfortunately the relative link leads to a 404 error in the documentation site. We're looking into this internally to try to find a way to use the relative link here, but for now this will have to be an absolute link.
I'll follow up here with any updates from my team about this.

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for implementing those changes. I made one additional minor style suggestion that's optional, but I posted a note in regards to the relative link used for the events explorer, as it currently leads to a 404 error in the documentation site.

**Review by emarsha94 — APPROVED** [review]
Thanks for making all of the changes! Added one last change but should be good to go after that.

**Review by emarsha94 — APPROVED** [review]

**Review by emarsha94 — CHANGES_REQUESTED** [review]
Hi team - Apologies for the confusion here. It looks like the PR is still missing a logo in .svg format. Can you please upload a logo to the PR? Thanks!

File: buoyant_cloud/README.md
Comment (anawishnoff) [inline buoyant_cloud/README.md:16]: Is this step for OAuth? I think it would be easier to direct the user to the Connect Accounts button on the tile, which will be right next to where this text is displayed :)

File: buoyant_cloud/README.md
Comment (anawishnoff) [inline buoyant_cloud/README.md:50]: The difference between email vs. contact may be confusing here, we may just want to list out one option.

File: buoyant_cloud/manifest.json
Comment (anawishnoff) [inline buoyant_cloud/manifest.json:13]: Could you please also add a screenshot of the Datadog dashboard here?

**Review by emarsha94 — APPROVED** [review]
Thanks so much for your time on this! Looks good - approving :)

File: buoyant_cloud/README.md
Comment (anawishnoff) [inline buoyant_cloud/README.md:28]:
```suggestion
3. Click `Remove`

Additionally, ensure that all API keys associated with this integration have been disabled by searching for the integration name on the [API Keys page](https://app.datadoghq.com/organization-settings/api-keys).
```
Adding one more comment for oauth-specific integrations. thanks!

**Review by anawishnoff — APPROVED** [review]

File: buoyant_cloud/manifest.json
Comment (bgoldberg122) [inline buoyant_cloud/manifest.json:8]: Ah, we need the uninstallation instructions here too:
```suggestion
    "configuration": "README.md#Setup",
    "uninstallation": "README.md#Uninstallation",
```

File: buoyant_cloud/README.md
Comment (buraizu) [inline buoyant_cloud/README.md:15]:
```suggestion
1. Click the **Connect Accounts** button on the tile to complete the OAuth flow.
```

File: buoyant_cloud/README.md
Comment (buraizu) [inline buoyant_cloud/README.md:28]:
```suggestion
3. Click **Remove**.
```

File: buoyant_cloud/README.md
Comment (buraizu) [inline buoyant_cloud/README.md:27]:
```suggestion
1. Browse to the [Buoyant Cloud Settings][4] page.
2. Click the kebab menu to the right of your Datadog org.
```

**Review by buraizu — APPROVED** [review]
Just a few final edits for formatting consistency, but otherwise this LGTM

File: .github/CODEOWNERS
Comment (bgoldberg122) [inline .github/CODEOWNERS:28]: Could you please add `makretplace-review as a CODEOWNER:
```suggestion
/buoyant_cloud/                          @BuoyantIO/buoys @DataDog/marketplace-review
```

**Review by bgoldberg122 — APPROVED** [review]

File: buoyant_cloud/manifest.json
Comment (anawishnoff) [inline buoyant_cloud/manifest.json:90]:
```suggestion
    "name": "Buoyant",
```
Sorry, should be the last thing - but this should by the company's name, not yours.

**Pull request merged**
